OpenLIFUData/OpenLIFUData.py

```diff
@@ -466,13 +547,69 @@ def onAddNewSubjectClicked(self, checked:bool) -> None:
                 #Update loaded subjects view
                 self.updateSubjectSessionSelector()
     
+    @display_errors
+    def getSelectedSubjectSession(self, index: qt.QModelIndex) -> Tuple[str, str]:
+        """ Returns the subject or session (name, id) currently selected in the SubjectSessionView """
+        
+        name = self.subjectSessionItemModel.itemFromIndex(index.siblingAtColumn(0)).text()
+        id = self.subjectSessionItemModel.itemFromIndex(index.siblingAtColumn(1)).text()
+        
+        return (name, id)
+
     @display_errors
     def onAddVolumeToSubjectClicked(self, checked:bool) -> None:
         volumedlg = AddNewVolumeDialog()
         returncode, volume_filepath, volume_name, volume_id = volumedlg.customexec_()
         if not returncode:
             return False
-        self.logic.add_volume_to_database(self.currentSubjectID, volume_id, volume_name, volume_filepath)
+        
+        currentIndex = self.ui.subjectSessionView.currentIndex()
+        current_subject = self.getSelectedSubjectSession(currentIndex)
+        self.logic.add_volume_to_database(current_subject[1], volume_id, volume_name, volume_filepath)
+
+    @display_errors
+    def onCreateNewSessionClicked(self, checked:bool) -> None:
+
+        currentIndex = self.ui.subjectSessionView.currentIndex()
+        current_subject = self.getSelectedSubjectSession(currentIndex)
+        subject_id = current_subject[1]
+        db_transducer_ids = self.logic.db.get_transducer_ids()
+        db_protocol_ids = self.logic.db.get_protocol_ids()
+        db_volume_ids = self.logic.db.get_volume_ids(subject_id)
```

Comment:
Can we add a check somewhere before these that `self.logic.db` is not `None`?

I know it's not supposed to be possible for it to be `None` because the button should be disabled and so on, but it still makes me a bit nervous since that assumes we don't ever violate those assumptions in the future ... maybe just raise a RuntimeError if self.logic.db is None?